metis-harvesting/src/test/java/eu/europeana/metis/harvesting/http/CompressedFileExtractorGzTest.java

```diff
@@ -0,0 +1,69 @@
+package eu.europeana.metis.harvesting.http;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
+import org.apache.commons.io.FileUtils;
+import org.junit.AfterClass;
+import org.junit.Test;
+
+public class CompressedFileExtractorGzTest {
+
+  private final static String DESTINATION_DIR = "src/test/resources/__files/";
+  private final static int XML_FILES_COUNT = 13;
+  private final static String FILE_NAME = "gzFile";
+  private final static String FILE_NAME2 = "gzFileWithCompressedGZFiles";
+  private final static String FILE_NAME3 = "gzFilesWithMixedCompressedFiles";
+  private final static String XML_TYPE = "xml";
+
+  @Test
+  public void shouldUnpackTheTarGzFilesRecursively() throws IOException {
+    CompressedFileExtractor.extractFile(Path.of(DESTINATION_DIR + FILE_NAME + ".tar.gz"),
+            Path.of(DESTINATION_DIR));
+    Collection<File> files = getXMLFiles(DESTINATION_DIR + FILE_NAME);
+    assertNotNull(files);
+    assertEquals(XML_FILES_COUNT, files.size());
+  }
+
+  @Test
+  public void shouldUnpackTheTarGzFilesRecursivelyWithCompressedXMLFiles() throws IOException {
+    CompressedFileExtractor.extractFile(Path.of(DESTINATION_DIR + FILE_NAME2 + ".tar.gz"),
```

Comment:
The point of this test is to unpack the TarGz files with compressed XML files. However, `FILE_NAME2` has a value String of `gzFileWithCompressedGZFiles`. 
This is is a bit misleading for me, do you want to test this with compressed XML files or you want to get XML files from compressed Gz files?

Reply:
No, the idea here is to test it with nested compressed files. So this is a gz file that itself contains other gz files inside it. Does that make sense?

Reply:
Yes, now I understand, thanks!